File: Strings/Shortest Common Supersequence.cpp
class Solution
{
    public:
    //Function to find length of shortest common supersequence of two strings.
    int shortestCommonSupersequence(char* X, char* Y, int m, int n)
    {
        //code here
        int dp[m+1][n+1];
        int i,j;
        for(i=0;i<m+1;i++){
            dp[i][0] = i;
        }
        for(j=0;j<n+1;j++){
            dp[0][j] = j;
        }
        
        for(i=1;i<m+1;i++){
            for(j = 1;j<n+1;j++){
                dp[i][j] = 0;
                if(X[i-1] == Y[j-1]){
                    dp[i][j] = 1+dp[i-1][j-1];
                }
                else{
                    dp[i][j] = 1+min(dp[i-1][j],dp[i][j-1]);
                }
            }
        }
        return dp[m][n];
    }
};

File: dynamicProgramming/Shortest Common Supersequence.py
class Solution:
    
    #Function to find length of shortest common supersequence of two strings.
    def shortestCommonSupersequence(self, X, Y, m, n):
        dp = [[0 for j in range(n+1)] for i in range(m+1)]
        for i in range(m+1):
            dp[i][0] = i
        for j in range(n+1):
            dp[0][j] = j
        for i in range(1,m+1):
            for j in range(1,n+1):
                if X[i-1] == Y[j-1]:
                    dp[i][j] = 1+dp[i-1][j-1]
                else:
                    dp[i][j] = 1+min(dp[i-1][j],dp[i][j-1])
        return dp[m][n]
